File: JAVA/MétodosEstáticos.java
public class Matematica {

    public static int fatorial(int n) {
        if (n == 0 || n == 1) {
            return 1;
        } else {
            return n * fatorial(n - 1);
        }
    }

    public static void main(String[] args) {
        System.out.println("Fatorial de 5 é: " + Matematica.fatorial(5));
        System.out.println("Fatorial de 7 é: " + Matematica.fatorial(7));
    }
}

File: GOLANG/MétodosEstáticos.go
package main

import "fmt"

type Matematica struct{}

func (m Matematica) Fatorial(n int) int {
    if n == 0 || n == 1 {
        return 1
    }
    return n * m.Fatorial(n-1)
}

func main() {
    mat := Matematica{}

    fmt.Printf("Fatorial de 5 é: %d\n", mat.Fatorial(5))
    fmt.Printf("Fatorial de 7 é: %d\n", mat.Fatorial(7))
}
